scanners/nodejs/utiltity.go

```diff
@@ -0,0 +1,352 @@
+package nodejs
+
+import (
+	"fmt"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/bitrise-io/bitrise-init/models"
+	"github.com/bitrise-io/bitrise-init/steps"
+	"github.com/bitrise-io/bitrise-init/utility"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type nodeVersionResult struct {
+	version string
+	file    string
+}
+
+type checkScriptResult struct {
+	scripts                    []string
+	hasBuild, hasLint, hasTest bool
+}
+
+// Detection
+func checkPackageJSON(searchDir string) (string, error) {
+	log.TPrintf("Checking package.json")
+	packageJsonPath := filepath.Join(searchDir, packageJson)
+	if exists := utility.FileExists(packageJsonPath); exists {
+		log.TPrintf("- %s - found", packageJsonPath)
+		return filepath.Rel(searchDir, packageJsonPath)
+	}
+
+	log.TPrintf("- %s - not found", packageJsonPath)
+	return "", fmt.Errorf("package.json not found")
+}
+
+var nodeVersionFiles = []string{".nvmrc", ".node-version", ".tool-versions"}
+
+func checkNodeVersion(searchDir string) nodeVersionResult {
+	log.TPrintf("Checking Node version")
+
+	result := nodeVersionResult{
+		version: "",
+		file:    "",
+	}
```

Comment:
You are saving this into a local variable but it is not used in the function apart from returning it at the end. You can just simply do 
```
return nodeVersionResult{}
```
and Go will intitialise the variables with their default values (which is "" for strings).